File: daq/slc/javalib/b2rc/src/b2rc/core/COPPERNode.java
package b2rc.core;

public class COPPERNode extends RCNode {

	static public final String TAG = "copper_node";

	private HSLB[] _hslb_v = new HSLB[4];
	private DataSender _sender = new DataSender();
	private String _module_class;

	public COPPERNode() {
	}

	public COPPERNode(String name) {
		super(name);
	}

	public DataSender getSender() {
		return _sender;
	}

	public HSLB getHSLB(int slot) {
		return _hslb_v[slot];
	}

	public String getModuleClass() {
		return _module_class;
	}

	public void setHSLB(int slot, HSLB hslb) {
		_hslb_v[slot] = hslb;
	}

	public void setModuleClass(String module_class) {
		_module_class = module_class;
	}

	@Override
	public String getTag() {
		return TAG;
	}

	@Override
	public String getSQLFields() {
		StringBuffer buf = new StringBuffer();
		buf.append(super.getSQLFields());
		for ( int i = 0; i < _hslb_v.length; i++ ) {
			buf.append(", hslb_id_" + i + " smallint");
		}
		buf.append(_sender.getSQLFields());
		return buf.toString();
	}

	@Override
	public String getSQLLabels() {
		StringBuffer buf = new StringBuffer();
		buf.append(super.getSQLLabels());
		for ( int i = 0; i < _hslb_v.length; i++ ) {
			if ( _hslb_v[i]!= null ) {
				buf.append(", hslb_id_" + i);
			}
		}
		buf.append(_sender.getSQLLabels());
		return buf.toString();
	}

	@Override
	public String getSQLValues() {
		StringBuffer buf = new StringBuffer();
		buf.append(super.getSQLValues());
		for ( int i = 0; i < _hslb_v.length; i++ ) {
			if ( _hslb_v[i]!= null ) {
				buf.append(", " + _hslb_v[i].getId());
			} else {
				buf.append(", -1");
			}
		}
		buf.append(_sender.getSQLValues());
		return buf.toString();
	}

}

File: daq/slc/src/node/src/COPPERNode.cc
#include "COPPERNode.hh"

#include "DataSender.hh"
#include "HSLB.hh"
#include "Host.hh"

#include <util/StringUtil.hh>

#include <sstream>

using namespace B2DAQ;

COPPERNode::COPPERNode(const std::string& name)
  : NSMNode(name, "copper_node"), _sender()
{
  _hslb_v[0] = _hslb_v[1] = _hslb_v[2] = _hslb_v[3] = NULL;
  _sender = new DataSender();
}

COPPERNode::~COPPERNode() throw()
{
  delete _sender;
}

const std::string COPPERNode::getSQLFields() const throw()
{
  std::stringstream ss;
  ss << NSMNode::getSQLFields();
  for (size_t i = 0; i < MAX_HSLBS; i++) {
    ss << ", hslb_id_" << i << " smallint";
  }
  ss << _sender->getSQLFields();
  return ss.str();
}

const std::string COPPERNode::getSQLLabels() const throw()
{
  std::stringstream ss;
  ss << NSMNode::getSQLLabels();
  for (size_t i = 0; i < MAX_HSLBS; i++) {
    ss << ", hslb_id_" << i;
  }
  ss << _sender->getSQLLabels();
  return ss.str();
}

const std::string COPPERNode::getSQLValues() const throw()
{
  std::stringstream ss;
  ss << NSMNode::getSQLValues();
  for (size_t i = 0; i < MAX_HSLBS; i++) {
    int id = (_hslb_v[i] != NULL) ? _hslb_v[i]->getID() : -1;
    ss << ", " << id;
  }
  ss << _sender->getSQLValues();
  return ss.str();
}

